src/contexts/Language/Language.tsx

```diff
@@ -0,0 +1,72 @@
+import {
+  createContext,
+  ReactChild,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+import { Language, Languages, LANGUAGES } from "../../constants/languages";
+import { useQueryParams } from "../../hooks/useQueryParams";
+
+const defaultLang: Language = Languages.TypeScript;
+
+const LOCAL_KEY = "preferred-language";
+const PARAM_KEY = "l";
+
+const isValidLang = (lang: Language | null) => lang && LANGUAGES.includes(lang);
+
+export const LanguageContext = createContext<
+  [Language, (language: Language) => void]
+>([defaultLang, () => {}]);
+
+export const useLanguage = () => useContext(LanguageContext);
+
+export function LanguageProvider({ children }: { children: ReactChild }) {
+  const langFromParams = useQueryParams().get(PARAM_KEY) as Language | null;
```

Comment:
```suggestion
  const langFromParams = useQueryParams().get(PARAM_KEY) as Language | void;
```

We tend to prefer `undefined` over null and not mixing between them.

Reply:
Makes sense. The annotation from the hook was `string | null` so really just wanted to cast the string portion. Wil update!